### ISSUE TITLE
have two channels for the website

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell.github.io#808]
as the following PR has been closed
- #804

i've been thinking about something that could be cool to allow
- probably the major part to enjoy a stable experience when looking at the website
- people like me who build `nushell` from source almost everyday to have access to bleeding edge documentation

## :grey_question: could it be possible to have two parallel channels for the website?
- :balance_scale: the **default one would build the final HTML pages based on the latest stable tag**
- :drop_of_blood: the other one could be **toggled on or off and would show up the latest `main`**

### implications
- :balance_scale: tag this repo once in a while when the version is bumped up
- :drop_of_blood: people could, after a documentation-changing PR over in the main `nushell`, apply the changes here with a single `nu make_docs.nu` and link to the associated `nushell` PR for completeness

**what do you think?** :yum: :wave:

**Comment by fdncred** [comment]
just for my two cents, i find it hard enough to keep one site up to date. so, i wouldn't be in favor of two unless we had a maintainer who was dedicated to keeping both running and up to date.

**Comment by amtoine** [comment]
> just for my two cents, i find it hard enough to keep one site up to date. so, i wouldn't be in favor of two unless we had a maintainer who was dedicated to keeping both running and up to date.
>
yep, of course, that is a very good point and a legitimate concern :thinking: 

in order to make a proper decision, **what would such a maintainer have to do precisely to achieve this?** :yum:

**Comment by sholderbach** [comment]
I have to admit that I don't really see the point having a separate documentation for the "nightly" version of what is on the main branch on nushell/nushell at the moment. The command docs should be accessible through `help` and long-form writing will always require extra work from the contributors and might take a few days to arrive separate from the implementation PR. Making this more complicated just reduces the chance that people contribute to the core docs.

Where I can see value in the extra work of adding extra channels on the page: If we would be to be able to have documentation in the state of specific version. When we get more and more serious users while still not being stable, people might choose to stay behind for a bit on a specific version. They might be interested in the rules and command semantics at their particular release. While it is not really scalable to provide active support (thorough replies in issues, contributors helping out in discord) for people behind the curve having the docs for passive support would probably go a long way. (also interesting as a time capsule for implementers)

**Comment by rgwood** [comment]
I don't think that this is worth the effort right now. It would make the website substantially more complicated, and the potential audience would be very small (people who build from `main` _and_ primarily use the website instead of `help`).

Could be worthwhile after v1.0 though.

**Comment by amtoine** [comment]
thanks a lot for all your feedback, very sensible things going on here :relieved: 


@sholderbach 
> Where I can see value in the extra work of adding extra channels on the page: If we would be to be able to have documentation in the state of specific version. When we get more and more serious users while still not being stable, people might choose to stay behind for a bit on a specific version. They might be interested in the rules and command semantics at their particular release. While it is not really scalable to provide active support (thorough replies in issues, contributors helping out in discord) for people behind the curve having the docs for passive support would probably go a long way. (also interesting as a time capsule for implementers)
>
so do you mean having one version of the website per release? for the `0.76.1`, `0.76.0`, ...?

@rgwood 
> I don't think that this is worth the effort right now. It would make the website substantially more complicated, and the potential audience would be very small (people who build from main and primarily use the website instead of help).
>
yep that's not my case, i just felt a bit sorry for the website to lag behind the latest `main` :yum: 

> Could be worthwhile after v1.0 though.
>
your call :wink: 

in the end guys
- if you feel this is not worth it, you can close the issue
- if you feel effort could be great in the direction of @sholderbach or that it can wait after the `1.0` as suggested by @rgwood, you can also leave it open

cheers :wave: :blush:
